### PR TITLE
Wip/superm1/synaptics mst fixes

### PR DESCRIPTION
**Pull request by superm1**
Type of pull request:

- [ ] New plugin (Please include [new plugin checklist](https://github.com/fwupd/fwupd/wiki/New-plugin-checklist))
- [x] Code fix
- [ ] Feature
- [ ] Documentation

### CODE REVIEW

File: plugins/synaptics-mst/fu-synaptics-mst-device.c [hunk 1]
```diff
@@ -1454,7 +1457,7 @@ fu_synaptics_mst_device_rescan(FuDevice *device, GError **error)
 	/* this is not a valid customer ID */
 	if ((self->board_id >> 8) == 0x0) {
 		fu_device_remove_flag(device, FWUPD_DEVICE_FLAG_UPDATABLE);
-		fu_device_set_update_error(device, "cannot update as CustomerID is unset");
+		g_debug("%s: cannot update as CustomerID is unset", fu_device_get_name(device));
```
Comment (hughsie) [inline plugins/synaptics-mst/fu-synaptics-mst-device.c:1460]: Not sure I get this. Isn't setting the update error entirely appropriate? Maybe an inhibit might be better?

File: plugins/synaptics-mst/fu-synaptics-mst-device.c [hunk 2]
```diff
@@ -1377,7 +1378,9 @@ fu_synaptics_mst_device_rescan(FuDevice *device, GError **error)
 	/* set up the device name and kind via quirks */
 	guid0 = g_strdup_printf("MST-%u", self->board_id);
 	fu_device_add_instance_id(FU_DEVICE(self), guid0);
-	name_parent = fu_device_get_name(FU_DEVICE(self));
+	parent = fu_device_get_parent(FU_DEVICE(self));
```
Comment (hughsie) [inline plugins/synaptics-mst/fu-synaptics-mst-device.c:1381]: Ohh wow, this was super broken! This commit lgtm.

File: plugins/synaptics-mst/fu-synaptics-mst-device.c (hunk 1 above)
Comment (superm1) [reply plugins/synaptics-mst/fu-synaptics-mst-device.c:1460]: Having the error means that it shows up in `get-devices` output, but users can't do anything about it.  Yeah I think inhibit will work fine here too.

Comment (hughsie) [reply plugins/synaptics-mst/fu-synaptics-mst-device.c:1460]: I thought we only showed them in get-updates if they were updatable unless we did the `--show-all` thing?

Comment (superm1) [reply plugins/synaptics-mst/fu-synaptics-mst-device.c]: Nope - take a look at the logic: https://github.com/fwupd/fwupd/blob/master/src/fu-util-common.c#L251

I think inhibiting will help it though.

File: plugins/synaptics-mst/fu-synaptics-mst-device.c [hunk 3]
```diff
@@ -1452,12 +1456,10 @@ fu_synaptics_mst_device_rescan(FuDevice *device, GError **error)
...
-	if ((self->board_id >> 8) == 0x0) {
-		fu_device_remove_flag(device, FWUPD_DEVICE_FLAG_UPDATABLE);
-		fu_device_set_update_error(device, "cannot update as CustomerID is unset");
-	} else {
-		fu_device_add_flag(device, FWUPD_DEVICE_FLAG_UPDATABLE);
-	}
+	if ((self->board_id >> 8) == 0x0)
+		fu_device_inhibit(device,
+				  "invalid-customer-id",
+				  "cannot update as CustomerID is unset");
```
Comment (hughsie) [inline plugins/synaptics-mst/fu-synaptics-mst-device.c:1462]: I think you want braces here, but tiny nit.

**Review by hughsie — DISMISSED** [review]

**Pull request merged**